qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/RecoveryIT.java

```diff
@@ -442,6 +444,43 @@ public void testCloseIndexDuringRollingUpgrade() throws Exception {
         }
     }
 
+
+    /**
+     * We test that a closed index makes no-op replica allocation only.
+     */
+    public void testClosedIndexReplicaAllocation() throws Exception {
```

Comment:
I think this test passed with the current behaviour. Can we make a small PR for this test only?